src/pty.rs

```diff
@@ -162,3 +173,42 @@ pub fn unlockpt(fd: &PtyMaster) -> Result<()> {
 
     Ok(())
 }
+
+
+/// Create a new pseudoterminal, returning the slave and master file descriptors
+/// in `OpenptyResult`
+/// (see [openpty](http://man7.org/linux/man-pages/man3/openpty.3.html)). 
+///
+/// If `winsize` is not `None`, the window size of the slave will be set to
+/// the values in `winsize`. If `termios` is not `None`, the pseudoterminal's
+/// terminal settings of the slave will be set to the values in `termios`.
+#[inline]
+pub fn openpty<'a, 'b, T: Into<Option<&'a Winsize>>, U: Into<Option<&'b Termios>>>(winsize: T, termios: U) -> Result<OpenptyResult> {
+    use std::ptr;
+
+    let mut slave: libc::c_int = -1;
+    let mut master: libc::c_int = -1;
+    let c_termios = match termios.into() {
+        Some(termios) => termios as *const Termios,
+        None => ptr::null() as *const Termios,
+    };
+    let c_winsize = match winsize.into() {
+        Some(ws) => ws as *const Winsize,
+        None => ptr::null() as *const Winsize,
+    };
+    let ret = unsafe {
+        libc::openpty(
+            &mut master as *mut libc::c_int,
+            &mut slave as *mut libc::c_int,
+            ptr::null_mut(),
+            c_termios as *mut libc::termios,
+            c_winsize as *mut Winsize)
+    };
+
+    let _ = try!(Errno::result(ret));
```

Comment:
Can't this just be `Errno::result(ret)?;` instead? Can you explain what this line is doing for me?